modules/nf-core/metamdbg/asm/main.nf

```diff
@@ -0,0 +1,58 @@
+process METAMDBG_ASM {
+    tag "${meta.id}"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/metamdbg:1.0--hdcf5f25_1':
+        'biocontainers/metamdbg:1.0--hdcf5f25_1' }"
+
+    input:
+    tuple val(meta), path(reads)
+    val(input_type)
+
+    output:
+    tuple val(meta), path("contigs.fasta.gz"), emit: contigs
+    tuple val(meta), path("metaMDBG.log")    , emit: log
+    path "versions.yml"                      , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    switch(input_type) {
+        case "hifi": input = "--in-hifi ${reads}"; break
+        case "ont": input = "--in-ont ${reads}"; break
+        default:
+            error("ERROR: input_type must be one of either 'hifi' or 'ont'.")
+            break
+    }
+    """
+    metaMDBG asm \\
+        --threads ${task.cpus} \\
+        --out-dir . \\
+        ${args} \\
+        ${input}
+
```

Comment:
Should we be deleting a `tmp` directory if it makes one and then is done with it?

Reply:
The tmp/ directory contains a load of intermediate files, including log files that may be useful for inspection - I thought it was best to leave them just in case, but I'm not opposed to just deleting it.

Reply:
I don't know the tool at all, so if it is potentially useful then keep it around. But if it is going to just take up a lot of space then probably best to cleanup.